### PR TITLE
New StorageClass parsing + new volume ID generation and parsing

### PR DESCRIPTION
**Pull request by amdabhad**
- Added changes for parsing new StorageClass parameters in the same PR

- Changes for new volume ID in VolumeCreate, ControllerPublishVolume, NodePublishVolume (No major changes to Delete or Unpublish were required)

**Note:**
- Removed function `generateNewSCVolID` and using the same old `generateVolID` with few modifications based on `bool isNewVolumeType` and some logic change

- Changes related to `GetVolIdMembers` function:
	* Using this function which does the job of spliting and getting indivisual members of volume ID, to avoid parsing differently in different places for different types of volume IDs (old with lenghth 3 or 4 and new with length 8)
	* Moved this function to `gpfs_util.go` - as now it used by both Controller and Node services
	* Renamed it to `getVolIDMembers` - to avoid go lint warning
	* Set `IsFilesetBased` to true, for fileset based volumes of old types as well as for new type for volume

- **New StorageClass example:**
```
Name:                  new-sc
IsDefaultClass:        No
Annotations:           <none>
Provisioner:           spectrumscale.csi.ibm.com
Parameters:            compression=True,storageClassType=Advanced,tier=Fast,volBackendFs=fs1
AllowVolumeExpansion:  <unset>
MountOptions:          <none>
ReclaimPolicy:         Delete
VolumeBindingMode:     Immediate
Events:                <none>

```

- **Example volume IDs from PV describe:**

```
* New type of volume:
VolumeHandle:      1;0;4708221959868304599;9E680B0A:61A4CDAC;ibm-spectrum-scale-csi-driver;;pvc-dabdeb6e-a909-46fb-a315-c3d5535df5ae;/ibm/fs1/spectrum-scale-csi-volume-store/.volumes/pvc-dabdeb6e-a909-46fb-a315-c3d5535df5ae

* Old - LW:
VolumeHandle:      0;0;4708221959868304599;9E680B0A:61A4CDAC;ibm-spectrum-scale-csi-driver;;;/ibm/fs1/spectrum-scale-csi-volume-store/.volumes/pvc-5f5091d6-bc21-415b-aa74-6cfd0d61deed

* Old - dependent:
VolumeHandle:      0;1;4708221959868304599;9E680B0A:61A4CDAC;ibm-spectrum-scale-csi-driver;;pvc-3e9e2146-f04f-4e42-99d8-55a658f434c8;/ibm/fs1/spectrum-scale-csi-volume-store/.volumes/pvc-3e9e2146-f04f-4e42-99d8-55a658f434c8

* Old - Independent:
VolumeHandle:      0;2;4708221959868304599;9E680B0A:61A4CDAC;ibm-spectrum-scale-csi-driver;;pvc-3afcd949-2044-460c-8634-0723e51a4107;/ibm/fs1/spectrum-scale-csi-volume-store/.volumes/pvc-3afcd949-2044-460c-8634-0723e51a4107
```

### CODE REVIEW

File: driver/csiplugin/gpfs_util.go [hunk 1]
```diff
@@ -390,3 +424,76 @@ func numberInSlice(a int, list []int) bool {
...
+		vIdMem.VolType = splitVid[1]
+		vIdMem.ClusterId = splitVid[2]
+		vIdMem.FsUUID = splitVid[3]
+		vIdMem.ConsistencyGroup = splitVid[4]
+		vIdMem.ApplicationName = splitVid[5]
+		vIdMem.FsetName = splitVid[6]
+		if vIdMem.StorageClassType == STORAGECLASS_CLASSIC {
+			if vIdMem.VolType == FILE_DIRECTORYBASED_VOLUME {
+				vIdMem.IsFilesetBased = false
+			} else {
```
Comment (madhuthorat) [inline driver/csiplugin/gpfs_util.go:487]: In future we will have BLOCK_xxx volume types, so we can have a check for volume type for fileset based volumes in else part as well.

File: driver/csiplugin/gpfs_util.go [hunk 2]
```diff
@@ -390,3 +424,76 @@ func numberInSlice(a int, list []int) bool {
...
+				vIdMem.IsFilesetBased = true
+			}
+		} else {
+			vIdMem.IsFilesetBased = true
```
Comment (madhuthorat) [inline driver/csiplugin/gpfs_util.go:491]: why not have check for volume type when storage class in not classic ?

File: driver/csiplugin/gpfs_util.go (hunk 1 above)
Comment (amdabhad) [reply driver/csiplugin/gpfs_util.go:487]: I think we can have code changes for block volume separately later? These conditions are sufficient for current design.

File: driver/csiplugin/gpfs_util.go (hunk 2 above)
Comment (amdabhad) [reply driver/csiplugin/gpfs_util.go:491]: because we have only one type of volume for advanced storage class.

**Pull request merged**